tests/oauth/test_exchange_token.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from respx import MockRouter
+
+from spylib.oauth import OfflineTokenModel, OnlineTokenModel, exchange_token
+
+exchange_token_params = [
+    (
+        dict(
+            access_token='offline-token',
+            scope='read_products,read_orders',
+        ),
+        OfflineTokenModel,
+    ),
+    (
+        dict(
+            access_token='online-token',
+            scope='write_products,read_orders',
+            expires_in=86399,
+            associated_user_scope='read_products,read_orders',
+            associated_user=dict(
+                id=902541635,
+                first_name='John',
+                last_name='Smith',
+                email='john@example.com',
+                email_verified=True,
+                account_owner=True,
+                locale='en',
+                collaborator=False,
+            ),
+        ),
+        OnlineTokenModel,
+    ),
+]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('token_dict,token_model', exchange_token_params)
+async def test_exchange_token(respx_mock: MockRouter, token_dict, token_model):
+    shop = 'example.myshopify.com'
+    code = 'code'
+    api_key = 'api-key'
+    api_secret_key = 'api-secret-key'
+
+    respx_mock.post(
+        f'https://{shop}/admin/oauth/access_token',
+        json__code=code,
+        json__client_id=api_key,
+        json__client_secret=api_secret_key,
```

Comment:
Is this equivalent to the dict version below, any difference in functionality?
```py
respx_mock.post(
        f'https://{shop}/admin/oauth/access_token',
        json={ 'code': code, 'client_id': api_key, 'client_secret': api_secret_key } 
       ).respond(
        json=token_dict,
    )

```
Can we include a comment with the link to https://lundberg.github.io/respx/api/#json for quick look up since it's not a commonly used syntax?

Reply:
@lishanl I'll add the doc link, good idea

There is a small difference I think. When passing the `json={}` the whole dict has to match exactly. With `json__` it only cares about that specific key. I guess I also like the `json__` as I don't have to use quotes - feels cleaner I guess